lib/syskit/exceptions.rb

```diff
@@ -483,6 +483,39 @@ def pretty_print(pp)
         end
     end
 
+    class ConflictingDeploymentAllocation < SpecError
```

Comment:
One good thing is to explicitly test the pretty_print method.

You can do it by creating the exception object and then

`text = PP.pp(exception, +"")`

Afterwards there's usually some sanitizing to do (`id:\d+` and `0x[0-9a-f]+` need to be removed because they vary from call to call)